gradio/components.py

```diff
@@ -237,23 +237,28 @@ def hash_url(self, url: str, chunk_num_blocks: int = 128) -> str:
             sha1.update(data)
         return sha1.hexdigest()
 
-    def hash_base64(self, base64_encoding: str, chunk_num_blocks: int = 128) -> str:
+    def _hash(self, base64_encoding: str | bytes, chunk_num_blocks: int = 128):
```

Comment:
The hashing functions should probably be moved into free functions in e.g. `utils.py` (or a new `hashing.py`) as they do not use `self` for anything.

Reply:
Used staticmethod where possible and filed this to track the wider refactor: https://github.com/gradio-app/gradio/issues/4270